send-transaction-service/src/send_transaction_service.rs

```diff
@@ -305,9 +352,67 @@ impl SendTransactionService {
             }
         });
 
+        if batched_transactions.is_empty() {
+            return result;
+        }
+
+        // Processing the transactions in batch
+        let addresses = leader_info
+            .as_ref()
+            .map(|leader_info| leader_info.get_leader_tpus(config.leader_forward_count));
+        let addresses = addresses
+            .map(|address_list| {
+                if address_list.is_empty() {
+                    vec![tpu_address]
+                } else {
+                    address_list
+                }
+            })
+            .unwrap_or_else(|| vec![tpu_address]);
+
+        let wire_transacions = transactions
+            .iter()
+            .filter(|(signature, _)| batched_transactions.contains(signature))
+            .map(|(_, transaction_info)| &transaction_info.wire_transaction as &[u8])
+            .collect::<Vec<&[u8]>>();
+
+        for address in &addresses {
+            if let Err(err) =
+                connection_cache::send_wire_transaction_batch(&wire_transacions, address)
+            {
+                warn!(
+                    "Failed to send transactions in batch to {}: {:?}",
+                    tpu_address, err
+                );
+            }
+        }
         result
     }
 
+    fn send_single_transaction<T: TpuInfo + std::marker::Send + 'static>(
+        tpu_address: &SocketAddr,
+        leader_info: &Option<T>,
+        config: &Config,
+        transaction_info: &TransactionInfo,
+    ) {
+        // non-batched send
+        let addresses = leader_info
+            .as_ref()
+            .map(|leader_info| leader_info.get_leader_tpus(config.leader_forward_count));
+        let addresses = addresses
+            .map(|address_list| {
+                if address_list.is_empty() {
+                    vec![tpu_address]
+                } else {
+                    address_list
+                }
+            })
```

Comment:
```suggestion
            .filter_map(|leader_info| {
                let address_list = leader_info
                    .get_leader_tpus(config.leader_forward_count));
                if address_list.is_empty() {
                    None
                } else {
                    Some(address_list)
                }
            })
```

Reply:
This changes the pre-existing logic -- it was using the input tpu_address if it did not get from get_leader_tpus.

Reply:
Oh -- you are not replacing the .unwrap_or_else(|| vec![&tpu_address]);. Neat!

Reply:
I will leave this as is for now -- it is not new code. the filter_map does not compile. Let me see what we can do to eliminate the double reference of tpu_address